src/arche/readers/items.py

```diff
@@ -39,10 +39,21 @@ def flat_df(self) -> pd.DataFrame:
     def process_df(df: pd.DataFrame) -> pd.DataFrame:
         # clean empty objects - mainly lists and dicts, but keep everything else
         df = df.applymap(lambda x: x if x or isinstance(x, numbers.Real) else np.nan)
-        if "_type" in df.columns:
-            df["_type"] = df["_type"].astype("category")
+        Items.categorize(df)
         return df
 
+    @staticmethod
+    def categorize(df: pd.DataFrame) -> pd.DataFrame:
+        if len(df) < 100:
+            return
+        for c in df.columns:
+            try:
+                if df[c].nunique(dropna=False) < 11:
```

Comment:
I didn't know about nunique

Reply:
It works faster than `value_counts`. At least on the first call.